### PR TITLE
MM-18281 Refactor "app/server_app_adapters.go" to use structured logging

### PR DESCRIPTION
**Pull request by Lumexralph**
#### Summary

Change the usages of mlog. methods with fmt.Sprintf to proper mlog.Field() calls in file app/server_app_adapters.go and remove the unused fmt package.

#### Ticket Link

[MM-18281](https://mattermost.atlassian.net/browse/MM-18281)

Fixes https://github.com/mattermost/mattermost-server/issues/12332

### CODE REVIEW

File: app/server_app_adapters.go [hunk 1]
```diff
@@ -67,7 +66,7 @@ func (s *Server) RunOldAppInitalization() error {
 	}
 
 	if htmlTemplateWatcher, err := utils.NewHTMLTemplateWatcher("templates"); err != nil {
-		mlog.Error(fmt.Sprintf("Failed to parse server templates %v", err))
+		mlog.Error("Failed to parse server templates ", mlog.Err(err))
```
Comment (iomodo) [inline app/server_app_adapters.go]: Why do we need the trailing space?

Comment (Lumexralph) [reply app/server_app_adapters.go]: @iomodo Thanks for your feedback, I'll work on it. There is actually no reason for the trailing space.

Comment (Lumexralph) [reply app/server_app_adapters.go]: @iomodo Feedback has been implemented.

**Comment by Lumexralph** [comment]
> I still see that there is
> 
> https://github.com/mattermost/mattermost-server/blob/08e2b25b89e0caf5a279faea125ab104604d7576/app/server_app_adapters.go#L133
> 
> without using mlog to handle apperror

@Willyfrog Thanks for the feedback, I used the structured logging for the `AppError` in one of my PRs but then had feedback it doesn't improve the logging so that led to leaving it as it is in this PR.

**Comment by Willyfrog** [comment]
> > I still see that there is
> > https://github.com/mattermost/mattermost-server/blob/08e2b25b89e0caf5a279faea125ab104604d7576/app/server_app_adapters.go#L133
> > 
> > without using mlog to handle apperror
> 
> @Willyfrog Thanks for the feedback, I used the structured logging for the `AppError` in one of my PRs but then had feedback it doesn't improve the logging so that led to leaving it as it is in this PR.

Can you point to me where someone asked you to not do it? I've been through your recent contributions (thanks by the way!) and only could find the exact oposite where you did change it into an mlog field (see https://github.com/mattermost/mattermost-server/pull/12256/files#diff-d58e1eaf6d955070cddf6cef92280749R161)

**Comment by Lumexralph** [comment]
> > > I still see that there is
> > > https://github.com/mattermost/mattermost-server/blob/08e2b25b89e0caf5a279faea125ab104604d7576/app/server_app_adapters.go#L133
> > > 
> > > without using mlog to handle apperror
> > 
> > 
> > @Willyfrog Thanks for the feedback, I used the structured logging for the `AppError` in one of my PRs but then had feedback it doesn't improve the logging so that led to leaving it as it is in this PR.
> 
> Can you point to me where someone asked you to not do it? I've been through your recent contributions (thanks by the way!) and only could find the exact oposite where you did change it into an mlog field (see https://github.com/mattermost/mattermost-server/pull/12256/files#diff-d58e1eaf6d955070cddf6cef92280749R161)

@Willyfrog I am wrong about my initial reason, I actually did change it in another PR you pointed me to and it was a different thing. I will work on it and update you. Thanks for your patience.

**Comment by Lumexralph** [comment]
@Willyfrog Feedback has been implemented, kindly review.

**Comment by Lumexralph** [comment]
> @Lumexralph can you update with master so we can close this?

Sure!, let me get to it.

**Comment by Lumexralph** [comment]
@Willyfrog Done

**Comment by jespino** [comment]
Merged! Thanks @Lumexralph! :dancer:

**Pull request merged**